contracts/core/OptionsPremiumPricer.sol

```diff
@@ -0,0 +1,141 @@
+//SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.7.3;
+
+import {IVolatilityOracle} from "../interfaces/IVolatilityOracle.sol";
+import {IPriceOracle} from "../interfaces/IPriceOracle.sol";
+import {ICompToken} from "../interfaces/ICompToken.sol";
+import {IERC20Detailed} from "../interfaces/IERC20Detailed.sol";
+import {Math} from "../libraries/Math.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {Welford} from "../libraries/Welford.sol";
+
+contract OptionsPremiumPricer {
+    using SafeMath for uint256;
+
+    /**
+     * Immutables
+     */
+    IVolatilityOracle public immutable volatilityOracle;
+    IPriceOracle public immutable priceOracle;
+    ICompToken public immutable cToken;
+
+    constructor(
+        address _volatilityOracle,
+        address _priceOracle,
+        address _cToken
+    ) {
+        require(_volatilityOracle != address(0), "!_volatilityOracle");
+        require(_priceOracle != address(0), "!_priceOracle");
+        require(_cToken != address(0), "!_cToken");
+        volatilityOracle = IVolatilityOracle(_volatilityOracle);
+        priceOracle = IPriceOracle(_priceOracle);
+        cToken = ICompToken(_cToken);
+    }
+
+    /**
+     * @notice Calculates the premium of the provided option using Black-Scholes
+     * References for Black-Scholes:
+       https://www.macroption.com/black-scholes-formula/
+       https://www.investopedia.com/terms/b/blackscholes.asp
+     * @param st is the strike price of the option
+     * @param t is the time to expiration in years
+     * @param isPut is whether the option is a put option
+     */
+    function getPremium(
+        uint256 st,
+        uint256 t,
+        bool isPut
+    ) external returns (uint256 premium) {
+        volatilityOracle.commit();
+
+        uint256 sp = priceOracle.latestAnswer().div(priceOracle.decimals());
+        uint256 v = volatilityOracle.vol();
+        uint256 rf = getAssetRiskFreeRate();
```

Comment:
todo: incorporate rf into black scholes pricing, unless we can ignore and just call [quoteAll](https://etherscan.io/address/0xCCdfCB72753CfD55C5afF5d98eA5f9C43be9659d#code) in keep3r volatility contract

Reply:
We can assume 0% rate for our own purposes since we're doing weekly options. We can revisit this in the future when we do longer dated options. So we can just remove out adding the interest component tbh